### PR TITLE
Ignore unreads in low priority rooms in the space panel

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/element-web/issues/16836

<hr>

This PR makes it so that the rooms that were set as `Low priority` and `Mentions & Keywords` won't cause a space to have a white dot indicating an unread message if they have unread messages. This is done to avoid the user being overwhlmed by unread messages in the space panel if they consider the room in which the message originated a `Low priority` 

---
Here's what your changelog entry will look like:

## ✨ Features
 * Ignore unreads in low priority rooms in the space panel ([\#6518](https://github.com/matrix-org/matrix-react-sdk/pull/6518)). Fixes vector-im/element-web#16836.

### CODE REVIEW

File: src/stores/notifications/SpaceNotificationState.ts
Comment (t3chguy) [inline src/stores/notifications/SpaceNotificationState.ts:81]: You shouldn't be accessing `tags` on the room object directly, instead make use of `RoomListStore::getTagsForRoom` given that the RLS does some other funkiness atop `.tags`

Comment (SimonBrandner) [reply src/stores/notifications/SpaceNotificationState.ts:81]: Ah, I didn't know that :+1:

File: src/stores/notifications/SpaceNotificationState.ts
Comment (t3chguy) [inline src/stores/notifications/SpaceNotificationState.ts:86]: This looks wrong, roomTags is a string array, not an object

image


```suggestion
            if (
                roomTags.includes(DefaultTagID.LowPriority) &&
                state.color === NotificationColor.Bold
            ) continue;
```

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/stores/notifications/SpaceNotificationState.ts
Comment (SimonBrandner) [reply src/stores/notifications/SpaceNotificationState.ts]: Sorry about that

**Review by t3chguy — APPROVED** [review]
code-wise looks sane, though am not a fan of the O(n^2) for each state find the respective room. We can probably be smarter there.

**Comment by SimonBrandner** [comment]
> code-wise looks sane, though am not a fan of the O(n^2) for each state find the respective room. We can probably be smarter there.

If this gets accepted by product, I think we could cache a list of low prio rooms and listen for tag updates to improve this

**Comment by daniellekirkwood** [comment]
It's not clear from the issue what the chosen solution was from a user perspective, any chance we can get a TLDR and some screenshots?

**Comment by SimonBrandner** [comment]
> It's not clear from the issue what the chosen solution was from a user perspective, any chance we can get a TLDR and some screenshots?

I've updated the description

**Comment by daniellekirkwood** [comment]
Thanks, @SimonBrandner 

It makes sense to me that users are not notified at a Space level with a dot by any messages in rooms that have been deemed by users of little value. 

Said differently... A Space will not have a dot for any messages that are in rooms with Low Priority, that are on Mute, or that are in Mentions & Keywords (where the message does not contain a mention or a keyword). 

While I think this approach makes sense, I'd like someone from the design org, and specifically with Spaces knowledge to share their thoughts here as there may be some subtle things I'm missing. Maybe @nadonomy or @niquewoodhouse ?

**Comment by t3chguy** [comment]
@daniellekirkwood one issue here, is marking a room "Low Priority" doesn't impact your notifications whatsoever at the Matrix level, so you can still get push notifications for those rooms. Might be confusing if you get a push or desktop notification but your space panel doesn't light up at all.

**Comment by SimonBrandner** [comment]
> @daniellekirkwood one issue here, is marking a room "Low Priority" doesn't impact your notifications whatsoever at the Matrix level, so you can still get push notifications for those rooms. Might be confusing if you get a push or desktop notification but your space panel doesn't light up at all.

If you get a push notificaiton, the space would light up. This is just up unread messages that did not cause a notification

**Comment by t3chguy** [comment]
> A Space will not have a dot for any messages that are in rooms with Low Priority, that are on Mute, or that are in Mentions & Keywords (where the message does not contain a mention or a keyword).

@SimonBrandner this says otherwise, its `or`

**Comment by daniellekirkwood** [comment]
I agree with @t3chguy - it would be odd to get a notification and not know what space it's in - if Low Priority rooms send notifications those would need to show at the space level.

**Comment by t3chguy** [comment]
Correction, a notification will cause a number rather than a dot

**Comment by daniellekirkwood** [comment]
Waiting for approval from Design. Will add labels and update the assignee.

**Review by niquewoodhouse — APPROVED** [review]

**Pull request merged**
